src/CMakeLists.txt

```diff
@@ -1,7 +1,8 @@
-cmake_minimum_required(VERSION 3.7)
+cmake_minimum_required(VERSION 3.15)
 
 add_library(jsonbuilder 
     JsonBuilder.cpp
+    JsonExceptions.cpp
```

Comment:
Note: Put the exception throwing functions in their own .cpp file so that they can be replaced at link time using normal linker techniques.